plugins/modules/cloud/misc/proxmox_kvm.py

```diff
@@ -853,6 +955,14 @@ def create_vm(module, proxmox, vmid, newid, node, name, memory, cpu, cores, sock
         kwargs['numa'] = kwargs['numa_enabled']
         del kwargs['numa_enabled']
 
+    # PVE api expects strings for the following params
+    if module.params['nameservers']:
+        kwargs['nameserver'] = ' '.join(module.param['nameservers'])
+    if module.params['searchdomains']:
+        kwargs['searchdomain'] = ' '.join(module.param['searchdomains'])
+    kwargs.pop('nameservers')
+    kwargs.pop('searchdomains')
```

Comment:
Considering that due to https://github.com/ansible-collections/community.general/blob/97ce7fe08be57b876e6f13d076b868d646374188/plugins/modules/cloud/misc/proxmox_kvm.py#L912 the current code probably won't work in case these options are not specified, let's do it as follows:
```suggestion
    # PVE api expects strings for the following params
    if 'nameservers' in module.params:
        nameservers = module.param.pop('nameservers')
        if nameservers:
            kwargs['nameserver'] = ' '.join(nameservers)
    if 'searchdomains' in module.params:
        searchdomains = module.param.pop('searchdomains')
        if searchdomains:
            kwargs['searchdomain'] = ' '.join(searchdomains)
```
(Could someone please actually run the module to see whether it still works, both with and without the new arguments?)

Reply:
> (Could someone please actually run the module to see whether it still works, both with and without the new arguments?)

Yes, going to test the code.